iphone/cli/commands/_build.js

```diff
@@ -1954,9 +1958,8 @@ iOSBuilder.prototype.validate = function validate(logger, config, cli) {
 						logger.error(__('Unable to find any Xcode installations that support iOS SDK %s.', this.iosSdkVersion) + '\n');
 						process.exit(1);
 					}
-				} else if (cli.argv.target === 'simulator' && !cli.argv['build-only']) {
```

Comment:
You can't do this. This line must remain.

If you're doing a sim build, running the app in a sim, and you didn't specify an iOS SDK version, then do nothing. Why? Because on https://github.com/garymathews/titanium_mobile/blob/74c48667224892db965fcca6b70cbf1fffcabffd/iphone/cli/commands/_build.js#L2012 when it calls `ioslib.simulator.findSimulators()`, it will determine the best iOS sim and Xcode version based on the inputs and return the results. If there isn't a selected iOS SDK version, then we use the latest associated with the Xcode version for the simulator it chose (https://github.com/garymathews/titanium_mobile/blob/74c48667224892db965fcca6b70cbf1fffcabffd/iphone/cli/commands/_build.js#L2043-L2046).

Reply:
> If there isn't a selected iOS SDK version, then we use the latest associated with the Xcode version for the simulator it chose.

That is what this change fixes. It should not use the latest Xcode as it may not be supported and it's ignoring my `xcode-select` selection. With this change, it will use a supported Xcode version and select the first listed simulator that is greater than our `minIosVersion`.

Reply:
OK. I'm not concerned with what version of Xcode is used. I'm very concerned if the iOS SDK changes depending on whether or not `--build-only` is specified.

I think part of the complexity is the iOS SDK version is passed in as a command line argument when I believe it should be a setting in the `tiapp.xml`.

I'm fine with this PR, but then I believe lines 2043-2046 become dead code.

I think we also need to fix this in ioslib. The Titanium build shouldn't need to be burdened with all this selecting/sorting logic. Care will need to be taken as to not repeat what got us in this situation.